pkg/controller/storage/capability/capability_controller.go

```diff
@@ -257,40 +234,20 @@ func (c *StorageCapabilityController) syncHandler(key string) error {
 	storageClass, err := c.storageClassLister.Get(name)
 	if err != nil {
 		// StorageClass has been deleted, delete StorageClassCapability and VolumeSnapshotClass
-		if errors.IsNotFound(err) {
-			if c.snapshotAllowed() {
-				err = c.deleteSnapshotClass(name)
```

Comment:
please split the snapshotclass handling to a separated controller. A controller should only do one thing. This is not your fault, but improve this in this PR.